temporalio/worker/_nexus.py

```
@@ -0,0 +1,471 @@
+
+    # Only call this if run() raised an error
+    async def drain_poll_queue(self) -> None:
+        while True:
+            try:
+                # Take all tasks and say we can't handle them
+                task = await self._bridge_worker().poll_nexus_task()
+                completion = temporalio.bridge.proto.nexus.NexusTaskCompletion(
+                    task_token=task.task.task_token
+                )
+                completion.error.failure.message = "Worker shutting down"
+                await self._bridge_worker().complete_nexus_task(completion)
+            except temporalio.bridge.worker.PollShutdownError:
+                return
+
+    # Only call this after run()/drain_poll_queue() have returned. This will not
+    # raise an exception.
+    async def wait_all_completed(self) -> None:
+        await asyncio.gather(*self._running_tasks.values(), return_exceptions=True)
+
+    # TODO(nexus-preview): stack trace pruning. See sdk-typescript NexusHandler.execute
+    # "Any call up to this function and including this one will be trimmed out of stack traces.""
+
+    async def _handle_cancel_operation_task(
+        self, request: temporalio.api.nexus.v1.CancelOperationRequest, task_token: bytes
+    ) -> None:
+        """
+        Handle a cancel operation task.
+
+        Attempt to execute the user cancel_operation method. Handle errors and send the
+        task completion.
+        """
+        ctx = CancelOperationContext(
+            service=request.service,
+            operation=request.operation,
+        )
+        temporal_operation_context.set(
+            _TemporalNexusOperationContext(
+                nexus_operation_context=ctx,
+                client=self._client,
+                task_queue=self._task_queue,
+            )
+        )
+        # TODO(nexus-prerelease): headers
+        try:
+            await self._handler.cancel_operation(ctx, request.operation_token)
+        except Exception as err:
```

So this is the case where I think we should consider catching `BaseException`. I think we need to swallow all user exceptions that can occur for any reason and respond back